apis/v1/controllers/addresses.controller.js

```diff
@@ -0,0 +1,83 @@
+// ---------------
+// Based Imports
+// ---------------
+const Address = require('../../schemas/address.schema');
+const { JSONResponse } = require('../../utilities/response.utility');
+
+const User = require('../../schemas/user.schema');
+// ---------------
+
+
+
+// get all Address
+exports.getAllAddress = async (req, res, next) => {
+    try {
+        const addresses = await Address.find();
+
+        // if(!addresses || addresses.length === 0) return JSONResponse.success(res, 'Success.', [], 200);
+        
+        JSONResponse.success(res, 'Success.', addresses, 200);
+    } catch (error) {
+        JSONResponse.error(res, 'Error.', error, 404);
+    }
+}
+
+
+// get address by id
+exports.getAddressById = async (req, res, next) => {
+    try {
+        const address = await Address.findById(req.params.id);
+
+        if(!address) throw new Error('Address not found');
+
+        JSONResponse.success(res, 'Success.', address, 200);
+    } catch (error) {
+        JSONResponse.error(res, 'Error.', error, 404);
+    }
+}
+
+
+
+// create address
+exports. createAddress = async (req, res, next) => {
+    try {
+        const address = await Address.create(req.body)
+        
```

Comment:
Have a check here to look for the user_id property and check if it is of the ObjectID type. 
There is a function on the mongoose object called isValidObjectId which you can use to check if the user_id is a valid objectId

Reply:
ok